packages/mui-material/src/styles/experimental_extendTheme.js

```diff
@@ -27,7 +27,16 @@ function assignNode(obj, keys) {
 }
 
 function setColor(obj, key, defaultValue) {
-  obj[key] = obj[key] || defaultValue;
+  if (typeof defaultValue === 'function') {
+    try {
+      const token = defaultValue();
+      obj[key] = obj[key] || token;
+    } catch (error) {
+      // ignore error: this could happen if the color manipulators cannot parse the value.
```

Comment:
Maybe show a warning message? At least so that people know that they won't be able to depend on CSS vars for some CSS vals.

Reply:
Rethinking about the DX, it might be better to show warnings for default tokens that are used to calculate other tokens and silently skip the errors for the tokens that are custom/independent.